lib/manageiq_foreman/lib/manageiq_foreman/paged_response.rb

```diff
@@ -0,0 +1,42 @@
+module ManageiqForeman
+  # like the WillPaginate collection
+  class PagedResponse
+    include Enumerable
+
+    attr_accessor :page
+    attr_accessor :total
+    attr_accessor :size # subtotal
+    attr_accessor :results
+    # per_page, search, sort
+    def initialize(json)
+      if json.kind_of?(Hash) && json["results"]
+        @results = json["results"]
+        @total   = json["total"].to_i
+        @size    = @results.size
+        @page    = json["page"]
+      else # Array
+        @results = json.kind_of?(Hash) ? [json] : json
```

Comment:
since we introduced `map!`, maybe we no longer need case 3.
Not sure if it will creep up when provisioning.